crypto/evp/p_lib.c

```diff
@@ -716,6 +748,8 @@ int EVP_PKEY_assign(EVP_PKEY *pkey, int type, void *key)
         return 0;
 
     pkey->pkey.ptr = key;
+    detect_low_level_engine(pkey);
```

Comment:
Wouldn't it be simpler do have `detect_low_level_engine()` return the detected ENGINE and do this here?

``` C
    pkey->engine = detect_low_level_engine();
```

Reply:
I was afraid of doing that as the low level engine then will fail providing EVP_PKEY pmeths and ameths. And that might have some unwanted consequences. See the other part of this change - I need special handling of the low-level engine keys in the int_ctx_new() - and that would not go away. So it would not be that easy.